providers/flagsmith/src/test/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProviderTest.java

```diff
@@ -0,0 +1,105 @@
+package dev.openfeature.contrib.providers.flagsmith;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.QueueDispatcher;
+import okhttp3.mockwebserver.RecordedRequest;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.MutableContext;
+import dev.openfeature.sdk.ProviderEvaluation;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class FlagsmithProviderTest {
+
+    public static MockWebServer mockFlagsmithClient;
+
+    final QueueDispatcher dispatcher = new QueueDispatcher() {
+        @SneakyThrows
+        @Override
+        public MockResponse dispatch(RecordedRequest request) {
+            if (request.getPath().startsWith("/flags/")) {
+                return new MockResponse()
+                    .setBody(readMockResponse("environment_flags_response.json"))
```

Comment:
This seems wrong to me - the response json you have below looks like a response to the `environment-document` endpoint, yet you're binding it to the `flags` endpoint?

Reply:
Yes agreed, it's been changed to the flags endpoint result